### ISSUE TITLE
Problems Arise

### ISSUE THREAD

**Issue by Unfellowed** [janiluuk/L4D2_Rage#1]
In Campaign game, Engineer class can not press Shift key in the next chapter. Thank you for updating this plugin!

**Comment by TheRealSpiritJoker** [comment]
I believe this is caused by the Event_LeftStartArea not being called in campaign which sets the RoundStarted bool to true.

if this is the case it will also allow infinite class changes throughout the level.

**Comment by janiluuk** [comment]
I added possible fix for this in new pr; https://github.com/janiluuk/L4D2_DLRTalents/pull/4 - Last bigger refactor before getting to the fun stuff :)

I will try to test that PR contents during weekend, let know if find any other bugs.

**Comment by Unfellowed** [comment]
Hello! I found some issues and I want to report them to you. Sorry my English is not good! :(
1. Soldier's attack speed does not increase when hitting Special Infected.
https://youtu.be/7M3YqWSCT9Y
2. Saboteur's movement speed is permanently increased.
https://youtu.be/qslDNYKE1-E

**Comment by janiluuk** [comment]
> Hello! I found some issues and I want to report them to you. Sorry my English is not good! :(
> 
> 1. Soldier's attack speed does not increase when hitting Special Infected.
>    https://youtu.be/7M3YqWSCT9Y
> 2. Saboteur's movement speed is permanently increased.
>    https://youtu.be/qslDNYKE1-E

Thanks for reporting! These have been addressed in latest beta branch.

**Comment by xamionex** [comment]
I'm not sure if this is just my version or config but Medic's speed changes once you press CTRL, but it doesn't feel like speed or velocity change it feels like you're changing the host timescale
This doesn't happen with other classes.

https://user-images.githubusercontent.com/57235791/226635890-3b731ca0-3b25-40ee-9534-3beb54ff1334.mp4

https://user-images.githubusercontent.com/57235791/226636329-4646288f-c7f0-4487-a96c-f4111db63e6d.mp4

**Comment by xamionex** [comment]
Ammo and upgrades reset on class pick
Video:
https://xb.petar.cc/zoqo8/NubugIca53.mp4

**Comment by xamionex** [comment]
And also engineer's menu sometimes doesn't even work. Just randomly doesn't open

https://user-images.githubusercontent.com/57235791/227388190-806c24b8-4bcb-4b58-9697-eb7c59f927de.mp4

**Comment by xamionex** [comment]
Saboteur's middle click also randomly chooses skills, this time it chose airstrike but another time I played it chose engineer's turret.
**Though I feel like this is just because I'm admin and can switch classes mid-game, but I haven't tested it before starting the round if it happens**
![image](https://user-images.githubusercontent.com/57235791/227388933-e4442ed3-0c84-4dfd-b245-9f17e0daa673.png)
EDIT: Tested without switching classes from round start, still airstrike on saboteur

**Comment by Unfellowed** [comment]
> //// dlr_talents.sp
//
// C:\Users\Acer\Downloads\addons\sourcemod\scripting\include\dlr\perks.inc(140) : warning 246: function SlotIdToSlotClassname returns an array but return type is not marked as an array
// C:\Users\Acer\Downloads\addons\sourcemod\scripting\include\dlr\perks.inc(144) : warning 246: function SlotIdToSlotClassname returns an array but return type is not marked as an array
// C:\Users\Acer\Downloads\addons\sourcemod\scripting\include\dlr\perks.inc(147) : warning 246: function SlotIdToSlotClassname returns an array but return type is not marked as an array
// C:\Users\Acer\Downloads\addons\sourcemod\scripting\include\dlr\commands.inc(34) : error 092: number of arguments does not match definition
// C:\Users\Acer\Downloads\addons\sourcemod\scripting\dlr_talents.sp(328) : error 092: number of arguments does not match definition
//
// 2 Errors.
//
// Compilation Time: 2.39 sec
// ----------------------------------------

> //// dlr_talents_plugin_extendedsight.sp
//
// C:\Users\Acer\Downloads\addons\sourcemod\scripting\dlr_talents_plugin_extendedsight.sp(143) : error 010: invalid function or declaration
//
// 1 Error.
//
// Compilation Time: 0.33 sec
// ----------------------------------------
